packages/ra-core/src/controller/input/ReferenceArrayInputController.tsx

```diff
@@ -232,7 +234,10 @@ export class UnconnectedReferenceArrayInputController extends Component<
                     'The value of ReferenceArrayInput should be an array'
                 );
             }
-            crudGetMany(reference, ids);
+            const idsToFetch = isInitialCall
+                ? ids
+                : difference(ids, this.props.input.value);
+            crudGetMany(reference, idsToFetch);
```

Comment:
if `idsToFetch` is empty, the action shouldn't even be dispatched.

Reply:
Yes, I will change this :)